scripts/update_version.scp

```diff
@@ -0,0 +1,176 @@
+# (at your option) any later version.
+#
+# t8code is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with t8code; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+
+# This script updates the version of the library. It fetches the latest tag on main
+# and checks if the latest commit message is of the format "Update Version" or "Updated the version to MAJOR.MINOR.PATCH".
+# If the latest commit message is of the format "Update Version", it updates the version to the next version, 
+# depending on the first argument given (MAJOR, MINOR or PATCH). The second argument is the branch name.
+# If the latest commit message is of the format "Updated the version to MAJOR.MINOR.PATCH", it prints the 
+# new version number. If the latest commit message is not of the correct format, it exits and returns 1. 
+
+#
+# This script must be executed from the scripts/ folder.
+#
+if [ `basename $PWD` != scripts ]
+then
+  if [ -d scripts ]
+  then
+    # The directory stack is automatically reset on script exit.
+    pushd scripts/ > /dev/null
+  else
+    echo ERROR: scripts/ directory not found.
+    exit 1
+  fi
+fi
+
+git fetch --tags
+# Get the current branch name from the second argument
+BRANCH_NAME=$2
+echo "The branch name is: ${BRANCH_NAME}"
+
+# Get the current version from the latest tag on main
+VERSION_TEXT=$(git describe --tags origin/main)
+
+
+MAJOR=0
+MINOR=0
+PATCH=0
+
+# Get the version number from the version text. 
+if [[ $VERSION_TEXT =~ v([0-9]+)\.([0-9]+)\.([0-9]+) ]]; then
+    MAJOR=${BASH_REMATCH[1]}
+    MINOR=${BASH_REMATCH[2]}
+    PATCH=${BASH_REMATCH[3]}
+fi
+
+echo "Latest version is: ${MAJOR}.${MINOR}.${PATCH}"
+
+# Checkout the branch
+git checkout ${BRANCH_NAME}
+
+
+
+# Get the latest commit message
+LATEST_COMMIT_MESSAGE=$(git log -1 --pretty=%B)
+
+# Check if the latest commit message is of the format "Updated the version to MAJOR.MINOR.PATCH"
+if [[ $LATEST_COMMIT_MESSAGE =~ Updated\ the\ version\ to\ ([0-9]+)\.([0-9]+)\.([0-9]+) ]]; then
+    CHECK_MAJOR=${BASH_REMATCH[1]}
+    CHECK_MINOR=${BASH_REMATCH[2]}
+    CHECK_PATCH=${BASH_REMATCH[3]}
+    echo "The latest commit message is: Updated the version to ${CHECK_MAJOR}.${CHECK_MINOR}.${CHECK_PATCH}"
+    exit 0
+# Check if the latest commit message contains the phrase "Update Version"
+elif [[ $LATEST_COMMIT_MESSAGE == *"Update Version"* ]]; then
+    echo "Start Version Update"
+else
+    echo "Do not update the version"
+    exit 1
+fi
+
+# Check if the latest commit is empty
+if ! git diff --quiet HEAD~1 HEAD; then
+    echo "The latest commit is not empty. Exiting."
+    exit 1
+fi
```

Comment:
Do we still need this? This is now checked in the other workflows right?

Reply:
I would completely remove all the commit message checking from this script, since this scripts sole purpose is to update the version. The checking should happen by the workflow.
This would also enable us to trigger a version update with the workflow dispatch from the github website